### PR TITLE
Synopsis test verify instrumentation

### PR DESCRIPTION
**Pull request by mrose17**
Auditor: @ayumi 

Fixes https://github.com/brave/browser-laptop/issues/4122
 
Test Plan:
1. mkdir ~/.brave-test-ledger
2. NODE_ENV=test npm start
3. open the payments panel, visit any site, go to another site after less than 8s
4. verify that `about:preferences#payments` shows the traffic, e.g.,

### CODE REVIEW

File: app/ledger.js [hunk 1]
```diff
@@ -1291,11 +1313,8 @@ var syncWriter = (path, obj, options, cb) => {
 
 var pathName = (name) => {
   var parts = path.parse(name)
-  var basePath = process.env.NODE_ENV === 'test'
-    ? path.join(process.env.HOME, '.brave-test-ledger')
-    : app.getPath('userData')
 
-  return path.join(basePath, parts.name + parts.ext)
+  return path.join(app.getPath('userData'), parts.name + parts.ext)
```
Comment (diracdeltas) [inline app/ledger.js]: please revert this change. tests can't use the userData directory. https://travis-ci.org/brave/browser-laptop/builds/161965938

Comment (mrose17) [inline app/ledger.js]: fine.

Comment (diracdeltas) [inline app/ledger.js]: this is probably ok as of 3c1e03c93c279319b3a75d5d1f91cca56a09b733 now that i look at it, but there is some weirdness going on with tests right now

**Comment by mrose17** [comment]
thanks. i wouldn't be surprised about their being some weirdness. (-;

it's not clear to me how to test stuff like this in an automated fashion.

@ayumi - your thoughts?

**Comment by ayumi** [comment]
Looks like https://github.com/brave/browser-laptop/pull/4189/commits/ffcac8570ec7b8a122ab41358276b7fddc8ffc77 "when the idle state changes, flush the timer" from https://github.com/brave/browser-laptop/pull/4186 is also in this PR

**Comment by mrose17** [comment]
one is built on top of the other, so we have to review, approve them in that order, i think.

File: app/ledger.js (hunk 1 above)
Comment (diracdeltas) [inline app/ledger.js]: i take that back, i think this is fine. test failure was intermittent.

**Comment by ayumi** [comment]
Branch base– cool– it would be helpful to note the base in the PR

I tried this PR and it works; lgtm

**Comment by mrose17** [comment]
thanks. let me get the other PR resolved then we can merge this one.

**Comment by mrose17** [comment]
superseded by https://github.com/brave/browser-laptop/pull/4194

**Pull request closed without merging**